scripts/convert_lit_checkpoint.py

```diff
@@ -200,19 +203,25 @@ def maybe_unwrap_state_dict(lit_weights: Dict[str, torch.Tensor]) -> Dict[str, t
 
 def check_conversion_supported(lit_weights: Dict[str, torch.Tensor]) -> None:
     weight_names = {wk.split(".")[-1] for wk in lit_weights}
-    # LoRA or QLoRA
-    if any("lora" in wn for wn in weight_names):
-        raise ValueError("Model weights must be merged using `lora.merge_lora_weights()` before conversion.")
     # adapter v2. adapter_bias will only be in adapter_v2
-    elif "adapter_bias" in weight_names:
+    if "adapter_bias" in weight_names:
         raise NotImplementedError("Converting models finetuned with adapter_v2 not yet supported.")
     # adapter. gating_factor is in adapter and adapter_v2
     elif "gating_factor" in weight_names:
         raise NotImplementedError("Converting models finetuned with adapter not yet supported.")
 
 
+def merge_lora_weights_for_conversion(
+    out_dir: Path, lit_weights: Dict[str, torch.Tensor], config: LoRAConfig
+) -> Dict[str, torch.Tensor]:
+    model = LoRAGPT(config)
+    model.load_state_dict(lit_weights)
```

Comment:
Before merging, you need to load 2 checkpoints, the pretrained weights and the lora weights. See for example how the generation script does it https://github.com/Lightning-AI/lit-gpt/blob/main/generate/lora.py#L110-L112